src/Lib.P2P.Tests/SwarmTest.cs

```diff
@@ -0,0 +1,1244 @@
+                using (var stream = await swarm.DialAsync(peerB, "/ipfs/id/1.0.0"))
+                {
+                    Assert.IsNotNull(stream);
+                    Assert.IsTrue(stream.CanRead);
+                    Assert.IsTrue(stream.CanWrite);
+                }
+            }
+            finally
+            {
+                await swarm.StopAsync();
+                await swarmB.StopAsync();
+            }
+        }
+
+        [TestMethod]
+        public void PeerDiscovered()
+        {
+            var swarm = new SwarmService {LocalPeer = self};
+            var peerCount = 0;
+            swarm.PeerDiscovered += (s, e) => { ++peerCount; };
+            swarm.RegisterPeerAddress("/ip4/127.0.0.1/tcp/4001/ipfs/QmdpwjdB94eNm2Lcvp9JqoCxswo3AKQqjLuNZyLixmCM1h");
+            swarm.RegisterPeerAddress("/ip4/127.0.0.2/tcp/4001/ipfs/QmdpwjdB94eNm2Lcvp9JqoCxswo3AKQqjLuNZyLixmCM1h");
+            swarm.RegisterPeerAddress("/ip4/127.0.0.3/tcp/4001/ipfs/QmdpwjdB94eNm2Lcvp9JqoCxswo3AKQqjLuNZyLixmCM1h");
+            swarm.RegisterPeerAddress("/ip4/127.0.0.1/tcp/4001/ipfs/QmdpwjdB94eNm2Lcvp9JqoCxswo3AKQqjLuNZyLixmCM1i");
+            swarm.RegisterPeerAddress("/ip4/127.0.0.2/tcp/4001/ipfs/QmdpwjdB94eNm2Lcvp9JqoCxswo3AKQqjLuNZyLixmCM1i");
+            swarm.RegisterPeerAddress("/ip4/127.0.0.3/tcp/4001/ipfs/QmdpwjdB94eNm2Lcvp9JqoCxswo3AKQqjLuNZyLixmCM1i");
+            swarm.RegisterPeer(new Peer {Id = "QmdpwjdB94eNm2Lcvp9JqoCxswo3AKQqjLuNZyLixmCM1j"});
+            swarm.RegisterPeer(new Peer {Id = "QmdpwjdB94eNm2Lcvp9JqoCxswo3AKQqjLuNZyLixmCM1j"});
+
+            Assert.AreEqual(3, peerCount);
+        }
+
+        [TestMethod]
+        public async Task IsRunning()
+        {
+            var swarm = new SwarmService {LocalPeer = self};
+            Assert.IsFalse(swarm.IsRunning);
+
+            await swarm.StartAsync();
+            Assert.IsTrue(swarm.IsRunning);
+
+            await swarm.StopAsync();
+            Assert.IsFalse(swarm.IsRunning);
+        }
+
+        [TestMethod]
+        public async Task Connect_PrivateNetwork()
+        {
+            var peerB = new Peer
+            {
+                AgentVersion = "peerB",
+                Id = "QmdpwjdB94eNm2Lcvp9JqoCxswo3AKQqjLuNZyLixmCM1h",
+                PublicKey =
+                    "CAASXjBcMA0GCSqGSIb3DQEBAQUAA0sAMEgCQQDlTSgVLprWaXfmxDr92DJE1FP0wOexhulPqXSTsNh5ot6j+UiuMgwb0shSPKzLx9AuTolCGhnwpTBYHVhFoBErAgMBAAE="
+            };
+            var swarmB = new SwarmService {LocalPeer = peerB, NetworkProtector = new OpenNetwork()};
+            await swarmB.StartAsync();
+            var peerBAddress = await swarmB.StartListeningAsync("/ip4/127.0.0.1/tcp/0");
+            Assert.IsTrue(peerB.Addresses.Count() > 0);
+
+            var swarm = new SwarmService {LocalPeer = self, NetworkProtector = new OpenNetwork()};
+            await swarm.StartAsync();
+            try
+            {
+                var remotePeer = await swarm.ConnectAsync(peerBAddress);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'remotePeer'.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)